hw4/phonebook/phonebook.c

```diff
@@ -0,0 +1,198 @@
+﻿#include <stdio.h>
+#include <time.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+#define ARRAY_SIZE 100
+#define RANGE 11
+typedef struct PhoneBookEntry {
+    char name[30];
+    char phone[30];
+} PhoneBookEntry;
+
+struct PhoneBookEntry GlobalEntry[ARRAY_SIZE];
+int currentRecord = 0;
+
+void clrscr()
+{
+    system("@cls||clear");
+}
+
+int addRecord()
+{
+    PhoneBookEntry entry;
+
+    printf("\nEnter name and surname: ");
+    scanf("%[^\n]%*c", entry.name);
+
+    printf("Enter phone number: ");
+    scanf("%[^\n]%*c", entry.phone);
+    GlobalEntry[currentRecord] = entry;
+    ++currentRecord;
+    printf("Record has been added successfully!");
+    return 0;
+}
+
+int printRecords()
+{
+    FILE* file = fopen("input.txt", "r");
+    if (file == NULL) {
+        printf("The file is empty!");
+        return 1;
+    }
+    PhoneBookEntry entry;
+    int linesRead = 0;
+    if (feof(file))
+    {
+        printf("\nThe file is empty!");
+        fclose(file);
+        return 0;
+    }
+    printf("\nThere are all records in file:");
+    while (!feof(file)) {
+        char* buffer = malloc(sizeof(char) * 30);
+        const int readBytes = fscanf(file, "%[^\n]%*c", buffer);
+        if (readBytes < 0) {
```

Comment:
```suggestion
        if (readBytes < 0) 
        {
```
В этом коде такого много, но я где-то читал, что отмечать несколько раз одну ошибку --- это пассивная агрессия :)